restauranty-service-impl/src/main/java/pl/taskyers/restauranty/service/impl/restaurants/RestaurantServiceImpl.java

```diff
@@ -0,0 +1,45 @@
+package pl.taskyers.restauranty.service.impl.restaurants;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import pl.taskyers.restauranty.core.data.restaurants.converters.RestaurantConverter;
+import pl.taskyers.restauranty.core.data.restaurants.dto.RestaurantDTO;
+import pl.taskyers.restauranty.core.data.restaurants.entity.Restaurant;
+import pl.taskyers.restauranty.core.error.exceptions.ValidationException;
+import pl.taskyers.restauranty.core.messages.container.ValidationMessageContainer;
+import pl.taskyers.restauranty.repository.addresses.AddressRepository;
+import pl.taskyers.restauranty.repository.restaurants.RestaurantRepository;
+import pl.taskyers.restauranty.service.impl.restaurants.validator.RestaurantDTOValidator;
+import pl.taskyers.restauranty.service.restaurants.RestaurantService;
+
+@Service
+@RequiredArgsConstructor
+public class RestaurantServiceImpl implements RestaurantService {
+    
+    private final RestaurantDTOValidator restaurantDTOValidator;
+    
+    private final RestaurantRepository restaurantRepository;
+    
+    private final AddressRepository addressRepository;
+    
+    @Override
+    public Restaurant addRestaurant(RestaurantDTO restaurantDTO) {
+        
+        ValidationMessageContainer validationMessageContainer = restaurantDTOValidator.validate(restaurantDTO);
+        if ( validationMessageContainer.hasErrors() ) {
+            throw new ValidationException(validationMessageContainer.getErrors());
+        }
+        
+        Restaurant toSave = RestaurantConverter.convertFromDTO(restaurantDTO);
+        if ( addressRepository.findByStreetAndZipCodeAndCityAndCountry(restaurantDTO.getAddress().getStreet(),
+                restaurantDTO.getAddress().getZipCode(), restaurantDTO.getAddress().getCity(),
+                restaurantDTO.getAddress().getCountry()).isPresent() ) {
+            toSave.setAddress(addressRepository.findByStreetAndZipCodeAndCityAndCountry(restaurantDTO.getAddress().getStreet(),
+                    restaurantDTO.getAddress().getZipCode(), restaurantDTO.getAddress().getCity(),
+                    restaurantDTO.getAddress().getCountry()).get());
```

Comment:
1. daj addressDTO do zmiennej, ew. kazde pole do zmiennej jak chcesz i tak potem ich uzywac
2. a jak nie znajdzie adresu to tez zapisujemy?